core/shared/src/main/scala/monocle/Lens.scala

```diff
@@ -251,6 +251,10 @@ object Lens {
   def codiagonal[S]: Lens[S \/ S, S] =
     PLens.codiagonal
 
+  /** creates a [[Lens]] from a Store comonad coalgebra */
+  def storeFCoalg[F[_], S, A](f: S => F[S])(implicit CS: ComonadStore[F, A]): Lens[S, A] =
```

Comment:
As long as `f` is a *comonad coalgebra*, classical lens laws should be satisfied (*PutGet*, *GetPut* and *PutPut*).